File: app/src/main/kotlin/fernandagallina/weatherapp/di/AppModule.java
package fernandagallina.weatherapp.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    Application app;

    public AppModule(Application application) {
        this.app = application;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return app;
    }
}

File: app/src/main/kotlin/fernandagallina/weatherapp/di/AppModule.kt
package fernandagallina.weatherapp.di

import android.app.Application

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class AppModule(private val app: Application) {

    @Provides
    @Singleton
    internal fun provideApplication(): Application {
        return app
    }
}
